src/webhooks/helius/helius.service.ts

```diff
@@ -72,13 +84,17 @@ export class HeliusService {
   }
 
   private async mintAction(enrichedTransaction: EnrichedTransaction) {
+    const mintAddress = new PublicKey(
+      enrichedTransaction.tokenTransfers.at(0).mint,
+    );
+    const nft = await this.metaplex.nfts().findByMint({ mintAddress });
     const payload = {
-      address: enrichedTransaction.tokenTransfers.at(0).mint,
-      candyMachineAddress: 'A3UgZc39HZbDiiDB24vjgNdmnV43RGznRavFkj5sJ68c',
-      collectionNftAddress: '68gsxWrLkMkwMznXM9qam7FvDK1SDBhVJZaQUpAwXL6k',
-      name: uuidv4(),
+      address: mintAddress.toBase58(),
+      candyMachineAddress: 'FRzbE9ENACT1ag8z4Q1JpQ5chU18GZq26Bxr8Vd71BDb',
+      collectionNftAddress: nft.collection.address.toBase58(),
+      name: nft.name,
       owner: enrichedTransaction.tokenTransfers.at(0).toUserAccount,
-      uri: 'uri',
+      uri: nft.uri,
     };
```

Comment:
Please consult with Athar on the best approach for getting the NFT name, URI, Collection NFT address, and the Candy Machine address.

If we know the CM we can easily figure out the Collection NFT

Reply:
@athar-d-reader

Reply:
we can use 
```
    const metadataPda = this.metaplex.nfts().pdas().metadata({mint});
    const info = await this.metaplex.rpc().getAccount(metadataPda);
    const metadata = toMetadata(toMetadataAccount(info));
```
with this we are making only 1 call for the required things and saving us 3 calls to rpc;
we can import ``toMetadata`` and ``toMetadataAccount`` from ``@metaplex-foundation/js``

Reply:
ty @athar-d-reader 

@d-reader-luka I'll resolve this by Monday and merge the PR in. No need for you to worry about it